gnovm/pkg/gnolang/realm.go

```diff
@@ -871,6 +908,135 @@ func (rlm *Realm) clearMarks() {
 	rlm.escaped = nil
 }
 
+// assertNoPrivateDeps ensures that the object is not private
+// it does not recursively check the values, but it does check recursively the types
```

Comment:
You cannot just check the types if you have interfaces; interfaces may contain values from any package within....